File: wagerproof-ios-native/Wagerproof/Features/Agents/Components/GlowingCardWrapper.swift
import SwiftUI

/// Native SwiftUI port of `components/agents/GlowingCardWrapper.tsx`. RN uses
/// the `react-native-animated-glow` library to draw a shifting multi-stop
/// halo around top-3 leaderboard avatars. We approximate the effect with a
/// `RoundedRectangle` `.stroke` + `.blur` ring.
///
/// FIDELITY-WAIVER #071: Same as `GlowAccentBar` — the animated color cycle
/// is dropped. The static brand-tinted halo still calls out the top three
/// leaderboard rows but doesn't pulse.
struct GlowingCardWrapper<Content: View>: View {
    let color: String
    var cornerRadius: CGFloat = 20
    let content: () -> Content

    init(color: String, cornerRadius: CGFloat = 20, @ViewBuilder content: @escaping () -> Content) {
        self.color = color
        self.cornerRadius = cornerRadius
        self.content = content
    }

    var body: some View {
        let (primary, secondary) = AgentColorPalette.gradient(for: color)
        content()
            .background(
                ZStack {
                    // Outer halo (large, soft).
                    RoundedRectangle(cornerRadius: cornerRadius + 4, style: .continuous)
                        .stroke(
                            LinearGradient(
                                colors: [primary.opacity(0.6), secondary.opacity(0.6)],
                                startPoint: .topLeading,
                                endPoint: .bottomTrailing
                            ),
                            lineWidth: 4
                        )
                        .blur(radius: 6)

                    // Inner ring (tight, vivid).
                    RoundedRectangle(cornerRadius: cornerRadius + 2, style: .continuous)
                        .stroke(
                            LinearGradient(
                                colors: [primary, secondary],
                                startPoint: .topLeading,
                                endPoint: .bottomTrailing
                            ),
                            lineWidth: 1
                        )
                        .blur(radius: 1)
                }
                .padding(-3)
            )
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/agents/components/GlowingCardWrapper.kt
package com.wagerproof.app.features.agents.components

import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.wagerproof.app.features.agents.shaded

/**
 * Static halo around top-3 leaderboard avatars — port of iOS `GlowingCardWrapper.swift`.
 *
 * FIDELITY-WAIVER #071: the animated multi-stop color cycle is dropped. A soft
 * outer ring + tight inner ring calls out the row without pulsing.
 */
@Composable
fun GlowingCardWrapper(
    color: Color,
    modifier: Modifier = Modifier,
    cornerRadius: Dp = 20.dp,
    content: @Composable () -> Unit,
) {
    val secondary = color.shaded(0.7)
    val softBrush = Brush.linearGradient(listOf(color.copy(alpha = 0.6f), secondary.copy(alpha = 0.6f)))
    val vividBrush = Brush.linearGradient(listOf(color, secondary))

    Box(modifier = modifier) {
        // Halo rings sit behind the content, bleeding 3dp past its bounds.
        Box(
            Modifier
                .matchParentSize()
                .padding((-3).dp)
                .blur(6.dp)
                .border(4.dp, softBrush, RoundedCornerShape(cornerRadius + 4.dp)),
        )
        Box(
            Modifier
                .matchParentSize()
                .padding((-3).dp)
                .blur(1.dp)
                .border(1.dp, vividBrush, RoundedCornerShape(cornerRadius + 2.dp)),
        )
        content()
    }
}
